training/CHAPEL_TUTORIAL.md

```diff
@@ -993,358 +1073,635 @@ Expected Output:
-into `upper` and `lower`, 2 distributed `uint` arrays.
-```Chapel
-use BlockDist, CommDiagnostics, Time;
+When you do a `forall` loop over a distributed array, each locale will handle the values that are local to it.
+So we can go about this 2 different ways:
+* We could loop over the `src` array, so the source values will always be local. We then copy our local values
+into a remote position of the `dst` array using a `DstAggregator`
+* Or we loop over the `dst` array, so destination values are local. We then want to copy remote values from
+the `src` array into our local `dst` array positions using a `SrcAggregator`
 
-// simplified symenty class
-class SymEntry {
-    type etype;
-    var a;
+When doing a `forall` loop over a distributed array, I like to think of it as doing the computation from the
+perspective of that array. For this example, we can accomplish our goal just as easily using source or destination
+aggregation. But in some cases one may be easier than the other.
 
-    proc init(len: int, type etype) {
-        this.etype = etype;
-        this.a = Block.createArray({0..#len}, etype);
-    }
+We'll use a [config variable](https://chapel-lang.org/docs/users-guide/base/configs.html)
+to switch between source and destination aggregation. When `UseDstAgg` is set to `true` (the default), it will use
+destination aggregation. To switch to source aggregation we need to add the flag `--UseDstAgg=false` to our execution
+command.
 
-    proc init(in a: [?D] ?etype) {
-        this.etype = etype;
-        this.a = a;
+```chapel
+use BlockDist, CopyAggregation;
+
+config const UseDstAgg = true;
+
+const dom = blockDist.createDomain({0..<6});
+
+// named src because this is the source we are copying from
+var src: [dom] int = [0, 1, 2, 3, 4, 5];
+
+// named dst because this is the destination we are copying to
+var dst: [dom] int;
+
+writeln("src: ", src);
+writeln("dst: ", dst);
+
+if UseDstAgg {
+    // when the destination is remote we use a dstAggregator
+    forall (s, i) in zip(src, 0..) with (var agg = new DstAggregator(int)) {
+      // locNum is which locale this loop iteration is executing on
+      var locNum = here.id;
+
+      // localSubDom is the chunk of the distributed arrays that live on this locale
+      var localSubDom = dom.localSubdomain();
+      
+      // we use a single writeln to avoid interleaving output from another locale 
+      writeln("\niteration num: ", i, "\n  on Locale: ", locNum,
+              "\n  on localSubDom: ", localSubDom, "\n  src[", i, "] is local",
+              "\n  dst[", (i + 3) % 6, "] is remote");
+    
+      // since dst is remote, we use a dst aggregator
+      // assignment without aggregation would look like:
+      // dst[ (i + 3) % 6 ] = s
+      agg.copy(dst[ (i + 3) % 6 ], s);
+    }
+    writeln();
+    writeln("src: ", src);
+    writeln("dst: ", dst);
+}
+else {
```

Comment:
Showing both the source and destination aggregator in one code example results in a lot of code.  What do you think about picking just one of them to start with (does Arkouda use one more than the other?) and then showing the second example in a details section or something like that?